lib.php

```diff
@@ -28,6 +28,12 @@
 define('XAPI_REPORT_SOURCE_FAILED', 'logstore_xapi_failed_log');
 define('XAPI_REPORT_SOURCE_HISTORICAL', 'logstore_standard_log');
 
+// Error types
+define('XAPI_REPORT_ERRORTYPE_NETWORKERROR', 101);
+define('XAPI_REPORT_ERRORTYPE_RECIPEERROR', 400);
+define('XAPI_REPORT_ERRORTYPE_AUTHERROR', 401);
+define('XAPI_REPORT_ERRORTYPE_LRSERROR', 500);
```

Comment:
It's good to add constants for these but as discussed on the call this morning, these aren't correct

Reply:
We'd need to update the corresponding error code when the failed to transform error occurs.

Reply:
@lzabo can we remove the changes to this function from this PR? I have made changes to this in another PR and we can address this there